timescaledb/tutorials/nfl-analytics/ingest-and-query.md

```diff
@@ -0,0 +1,283 @@
+    team_name TEXT,
+    team_abbreviation TEXT,
+    time_zone TEXT
+)
+
+```
+
+Add indexes to the `tracking` table to improve query performance:
+
+```sql
+CREATE INDEX idx_gameid ON tracking (gameid);
+CREATE INDEX idx_playerid ON tracking (player_id);
+CREATE INDEX idx_playid ON tracking (playid);
+```
+
+**Create hypertable from `tracking` table**
+```sql
+/*
+tracking: name of the table
+time: name of the timestamp column
+*/
+SELECT create_hypertable('tracking', 'time');
+```
+
+## Ingest data from CSV files
+
+There are three separate CSV files for game, player, and play tables. For the tracking table, you need to
+import data from 17 CSV files (one file for each week of the season).
+
+You can use a Python script that uses psycopg2's `copy_expert` function to ingest the data:
+
+```python
+import config
+import psycopg2
+
+# connect to the database
+conn = psycopg2.connect(database=config.DB_NAME, 
+                        host=config.HOST, 
+                        user=config.USER, 
+                        password=config.PASS, 
+                        port=config.PORT)
+
+# insert CSV file into given table
+def insert(csv_file, table_name):
+    cur = conn.cursor()
+    copy_sql = """
+            COPY {table} FROM stdin WITH CSV HEADER
+            DELIMITER as ','
+            """.format(table=table_name)
+    with open(csv_file, 'r') as f:
+        cur.copy_expert(sql=copy_sql, file=f)
+        conn.commit()
+        cur.close()
+
+print("Inserting games.csv")
+insert("data/games.csv", "game")
+
+print("Inserting plays.csv")
+insert("data/plays.csv", "play")
+
+print("Inserting players.csv")
+insert("data/players.csv", "player")
+
+print("Inserting stadium_info.csv")
+insert("data/stadium_info.csv", "stadium_info")
+
+print("Inserting scores.csv")
+insert("data/scores.csv", "scores")
+
+# iterate over each week's CSV file and insert them
+for i in range(1, 18): 
+    print("Inserting week{i}".format(str(i)))
+    insert("data/week{i}.csv".format(i=i), "tracking")
+
+conn.close()
+```
+
+## Run your first query
+
+Now that you have all the data ingested, let's run your first aggregation query 
+and examine the results. For most of the example queries that follow throughout 
+the tutorial, you'll need to aggregate data from the `tracking` table, which 
+contains multiple rows per player for each play (because the data is sampled 
+multiple times per second during each play)
+
+### Number of yards run in game for passing plays, by player and game
+
+This query sums all yards for each player in every game. You can then 
+join that on the `player` table to get player details.
+
+```sql
+SELECT t.player_id, p.display_name, SUM(dis) AS yards, t.gameid 
+FROM tracking t
+LEFT JOIN player p ON t.player_id = p.player_id 
+GROUP BY t.player_id, p.display_name, t.gameid 
+ORDER BY t.gameid ASC, yards DESC;
+```
+Your data should look like this:
+
+|player_id |        display_name         |       yards        |   gameid   |
```

Comment:
You can remove the padding from the table (although I admit it looks prettier in raw like this!)

Reply:
Yeah, this is honestly just a copy/paste from the tool. I didn't feel like taking the time to _remove_ it. 😂